pkg/kubelet/apis/config/types.go

```diff
@@ -123,6 +123,7 @@ type KubeletConfiguration struct {
 	// tlsPrivateKeyFile is the file containing x509 private key matching tlsCertFile
 	TLSPrivateKeyFile string
 	// TLSCipherSuites is the list of allowed cipher suites for the server.
+	// Note that TLS 1.3 ciphersuites are not configurable.
```

Comment:
would it make sense to prevent setting TLSCipherSuites and TLSMinVersion == "VersionTLS13" in `kubelet/app/server.go#InitializeTLS`?

Reply:
While this check would be correct, I personally like sticking as close to the official tls behavior as possible. One use-case might be K8s variations that have replaced the standard tls library.

I see it as an ease of (mis-)use vs. simplicity and aligned behavior argument. But I hold no strong opinion either way.

Reply:
at the very least, we should warn when setting a combination that won't work as expected as far as we know

Reply:
If we want to verify that the configuration matches the user's expectation, then we'd need to enforce that `TLSCipherSuites` _always_ contains all [defaultCipherSuitesTLS13](https://cs.opensource.google/go/go/+/master:src/crypto/tls/cipher_suites.go;l=345). As mentioned, we always need to do that since the kubelet does not provide a way to set `tls.Config.MaxVersion`, so TLS1.3 is always part of the enabled versions. 

Enforcing this will probably break existing KubeletConfigurations. So a warning is probably the best way forward. 

Which check do we want? Only warn when TLSMinVersion == "VersionTLS13" or the more restrictive check I proposed? 
After we have a decision, I will go ahead and implement it.

Reply:
I would start with warning when TLSMinVersion == "VersionTLS13" and TLSCipherSuites is non-empty